### PR TITLE
Speed up docker workflow osx

### PR DESCRIPTION
**Pull request by darrint**
I need someone on OSX to try this:

`dkgg watch-tasks watch`

Note how long it takes to build.

Save a few `.js` files. Note how long these builds take. Usually the first build takes a little longer than subsequent builds once `watch` gets caches warmed up.

`dk initscratchdata`

This is a one time thing. It will never be needed again unless you roast your docker storage somehow.

`dk optimizenode`

`dkgg watch-tasks watch`

Note build time.

Save a few `.js` files. If this worked then builds should be an order of magnitude faster.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -21,6 +21,7 @@ static/redirect.css
 
 # Gulp stuff
 gulp/node_modules/
+gulp/node_modules
```
Comment (aspidites) [inline .gitignore]: Was it not enough to just remove the slash? Seems weird to have to ignore both

File: dk2/dockerdev.sh [hunk 2]
```diff
@@ -42,6 +43,11 @@ initmysqldata() {
     docker create --name mysqldata darrint/mysqldata true
 }
 
+initscratchdata() {
+    docker build -t darrint/scratchdata scratchdata
```
Comment (aspidites) [inline dk2/dockerdev.sh]: Is this going to work with your name being referenced directly?

**Comment by koonsies** [comment]
A note about the steps listed on the PR.. You want to do the test on a separate branch first (probably quality-control) THEN checkout this branch and move forward. Otherwise, you receive the following:

Error response from daemon: no such id: scratchdata
  0.13s user 0.01s system 82% cpu 0.169 total

File: dk2/dockerdev.sh (hunk 2 above)
Comment (darrint) [inline dk2/dockerdev.sh]: Yes. In the future "darrint" should change to the hostname of a server where we can push and pull filesystem images.

File: .gitignore (hunk 1 above)
Comment (darrint) [inline .gitignore]: Didn't think that through. Git didn't ignore the symlink (which doesn't resolve locally) without the /.

**Comment by misterjonscott** [comment]
I'm seeing 'test' complete in ~3s, 'dev-bundle' in ~7s and 'lint' in ~11s, with <1s benefit after running 'optimizenode'

Discussed with @darrint and ready to assist with further testing.

**Comment by darrint** [comment]
Rats.

**Comment by darrint** [comment]
The command given at the end of the doc is approaching acceptable performance. In the future I can try some other approaches.

**Pull request closed without merging**